src/build.cc

```diff
@@ -388,7 +388,10 @@ void Plan::EdgeFinished(Edge* edge) {
   edge->outputs_ready_ = true;
 
   // See if this job frees up any delayed jobs
-  ResumeDelayedJobs(edge);
+  if (directly_wanted)
+    ResumeDelayedJobs(edge);
```

Comment:
I was thinking it was different because of the recursion, but I think you're right. Updated.